rules.tf

```
@@ -185,6 +185,28 @@ resource "aws_wafv2_web_acl" "default" {
                   content {}
                 }
 
+                dynamic "headers" {
+                  for_each = lookup(field_to_match.value, "headers", null) != null ? [field_to_match.value.headers] : []
```

I think this expression (and similar below) can be shortened:
```suggestion
                  for_each = compact([lookup(field_to_match.value, "headers", null)])

```

---

Ah, right, it doesn't work with lists of objects.
Also, I see that this pattern is already present in the code.